### ISSUE TITLE
java.lang.NoClassDefFoundError: RJavaClassLoader issue with calling R from Java in Eclipse

### ISSUE THREAD

**Issue by princeBruce** [s-u/rJava#97]
Currently I'm trying to use JRI to call a R script from my .java file. With the necessary jars being imported, I'm able to run simple R scripts with the Rengine object and read the results back with REXP object.

However, when I add a new line, which is **library("xlsx")**, in the same R script, java will throw java.lang.NoClassDefFoundError: RJavaClassLoader error. 

Does anyone have ideas regarding this issue? Thanks a lot.

**Comment by Albluca** [comment]
I think the problem is that xlsx relies on rJava. When you load the package it will try starting the JVM and add its jar the class path. However, since a JVM is already running it fails to do so and the jar of xlsx is not properly included.

I think that if you manually add the jar used by xlsx with .jaddClassPath() is should work. A quick workaround would also be to use readxl instead of xlsx.

**Comment by s-u** [comment]
When you start JRI, please make sure you include `rJava/java` on the `CLASSPATH` (the full path is obtainable by `system.file("java", package="rJava")`). As @Albluca pointed out, rJava assumes that it can bootstrap the class loader but if the JVM is already running it cannot change the `CLASSPATH` to set it to its own class loader.

**Comment by princeBruce** [comment]
Thanks for your responses.  @Albluca @s-u 

Since I use **Eclipse** as the IDE on **MAC**, here are a couple of things I have tried with the CLASSPATH.

1, I involved several .jar files (JRI 2.jar, JRIEngine 2.jar, and REngine 2. jar ) in Eclipses's build path (Project>Properties>java build path), though I'm not quite sure about where I got those .jar files some time back.

2, I type system.file("java", package="rJava") in R and find the full path as "/Library/Frameworks/R.framework/Versions/3.2/Resources/library/rJava/java", and then I add this path as a variable CLASSPATH in both Preferences>Java>Build path>classpath variables and Run>Run configurations>Environment. By doing this I thought it is equivalent to appending this string to the environment variable classpath. But when I type echo $classpath in shell, it does not show anything.

With above trials, it still gives the same error and I think I must have missed something. I also notice that rjava/java folder only contains .class and .java files, not .jar. Is there a standard way of including it in classpath when using Eclipse?
